components/proxy/src/main/java/com/hotels/styx/routing/config/RoutingObjectFactory.java

```diff
@@ -42,15 +43,31 @@
  * Builds a routing object based on its actual type.
  */
 public class RoutingObjectFactory {
-    private static final Map<String, HttpHandlerFactory> BUILTIN_HANDLERS = ImmutableMap.<String, HttpHandlerFactory>builder()
-            .put("StaticResponseHandler", new StaticResponseHandler.Factory())
-            .put("ConditionRouter", new ConditionRouter.Factory())
-            .put("InterceptorPipeline", new HttpInterceptorPipeline.Factory())
-            .put("ProxyToBackend", new ProxyToBackend.Factory())
-            .build();
+    public static final ImmutableMap<String, Schema.FieldType> BUILTIN_HANDLER_SCHEMAS;
+    private static final ImmutableMap<String, HttpHandlerFactory> BUILTIN_HANDLER_FACTORIES;
+    private static final String STATIC_RESPONSE = "StaticResponseHandler";
+    private static final String CONDITION_ROUTER = "ConditionRouter";
+    private static final String INTERCEPTOR_PIPELINE = "InterceptorPipeline";
+    private static final String PROXY_TO_BACKEND = "ProxyToBackend";
+
+    static {
+        BUILTIN_HANDLER_FACTORIES = ImmutableMap.<String, HttpHandlerFactory>builder()
```

Comment:
Looks like we can just use
```
BUILTIN_HANDLER_FACTORIES = ImmutableMap.of(
                STATIC_RESPONSE, new StaticResponseHandler.Factory(),
                CONDITION_ROUTER, new ConditionRouter.Factory(),
                INTERCEPTOR_PIPELINE, new HttpInterceptorPipeline.Factory(),
                PROXY_TO_BACKEND, new ProxyToBackend.Factory());
```
instead of the builder here.
Probably the same for `BUILTIN_HANDLER_SCHEMAS` as well.

Reply:
`ImmutableMap.of` takes maximum of four or five values. So after one (or two) more we'd have to revert back to builder. Therefore I'd leave it like this.